src/main/java/com/es/phoneshop/model/product/Product.java

```diff
@@ -66,6 +68,13 @@ public BigDecimal getPrice() {
 
     public void setPrice(BigDecimal price) {
         this.price = price;
+        if (priceHistory == null) {
```

Comment:
Not optimal mechanism, you inserted some business logic into setter - bad approach, but it's just hint, lets leave it as is.